components/builder-minio/habitat/config_install/minio-migration.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+function _hab_exec() {
+  pkg=$1
+  shift
+  hab pkg exec "$pkg" -- "$@"
+}
+
+function _aws() {
+  _hab_exec 'core/aws-cli' aws --endpoint-url "{{cfg.env.MINIO_ENDPOINT}}" "$@"
+}
+
+function _curl() {
+  _hab_exec 'core/curl' curl "$@"
+}
+
+function _jq() {
+  _hab_exec 'core/jq-static' jq "$@"
+}
+
+function download_bucket_objects() {
+  echo "DOWNLOADING objects from the MinIO that we are migrating from"
+  if ! _aws s3 sync "$s3_url" "$WAYPOINT"; then
+    echo "ERROR: download of objects FAILED"
+    exit 8
+  fi
+  return 0
+}
+
+function _ensure_bucket_exists() {
+  if ! _aws s3 ls "$MINIO_BUCKET" &>/dev/null; then
+    _aws s3 mb "$s3_url"
+  fi
+}
+
+function upload_bucket_objects() {
+  _ensure_bucket_exists
+  echo "UPLOADING objects to the MinIO that we are migrating to"
+  if ! _aws s3 sync "$WAYPOINT" "$s3_url"; then
+    echo "ERROR: upload of objects FAILED"
+    exit 9
+  fi
+  return 0
+}
+
+function is_migration_from_removed_fs_backend_needed() {
+  echo "CHECKING if migration from removed fs backend is needed"
+  if [[ -f /hab/svc/builder-minio/data/.minio.sys/format.json ]]; then
+    format_value=$(_jq -r '.format' /hab/svc/builder-minio/data/.minio.sys/format.json)
+    if [[ "${format_value}" == 'fs' ]]; then
+      return 0
+    fi
+  fi
+  return 1
+}
+
+function minio_health_live_check() {
+  for ((n = 0; n < 20; n++)); do
+    curl_http_code_args=(-fs -o /dev/null -w "%{http_code}" --retry 4 --retry-delay 1)
+    code=$(_curl "${curl_http_code_args[@]}" "$MINIO_ENDPOINT/minio/health/live")
+    if [[ $code == 200 ]]; then
+      return 0
+    else
+      sleep .5
+    fi
+  done
+  return 1
+}
+
+function minio_stop() {
+  for ((n = 0; n < 20; n++)); do
+    if pgrep minio >/dev/null; then
+      pkill minio >/dev/null
```

Comment:
this might be emitting to stderr

Reply:
Now redirecting both with &> instead of >

Reply:
Leaving the &> on these but the terminating output we had noticed was really coming from the minio server process so I've gotten that properly redirecting now too.